src/test/incremental/hashes/while_loops.rs

```diff
@@ -191,7 +191,7 @@ pub fn change_continue_label() {
 }
 
 #[cfg(not(cfail1))]
-#[rustc_clean(cfg="cfail2", except="HirBody, mir_built")]
+#[rustc_clean(cfg="cfail2", except="typeck_tables_of, HirBody, mir_built, optimized_mir")]
```

Comment:
Hmm, `while true` isn't being optimized to be the same as `loop` here. This is probably not blocking since we lint against `while true`. cc @oli-obk @wesleywiser

Reply:
So this would be (in HIR terms):
```rust
loop {
    match DropTemps(true) {
        true => $body,
        _ => break,
    }
}
```

Reply:
The typeck_tables_of bits are fixed now at least. The incremental behavior is the same for `while` and `while let` loops now which feels at least consistent.